File: tests/fixtures/skeleton/sample.swift
// Sample Swift fixture for Skeletonizer coverage tests.
import Foundation

struct User {
    let id: String
    let name: String
    let email: String
}

protocol UserRepository {
    func getUser(id: String) -> String?
}

class UserService: UserRepository {
    private let dbPath: String

    init(dbPath: String) {
        self.dbPath = dbPath
    }

    func getUser(id: String) -> String? {
        let raw = try? String(contentsOfFile: dbPath)
        // BODY_SENTINEL_DO_NOT_LEAK
        return raw
    }
}

func formatUser(user: User) -> String {
    // BODY_SENTINEL_DO_NOT_LEAK
    return "\(user.name) <\(user.email)>"
}

File: tests/fixtures/skeleton/sample.java
// Sample Java fixture for Skeletonizer coverage tests.
package com.example.skeleton;

import java.nio.file.Files;
import java.nio.file.Path;
import java.util.Optional;

public class UserService {
    private final String dbPath;

    public UserService(String dbPath) {
        this.dbPath = dbPath;
    }

    public Optional<String> getUser(String id) throws Exception {
        String raw = new String(Files.readAllBytes(Path.of(dbPath)));
        // BODY_SENTINEL_DO_NOT_LEAK
        return Optional.of(raw);
    }

    public static String formatUser(String name, String email) {
        // BODY_SENTINEL_DO_NOT_LEAK
        return name + " <" + email + ">";
    }
}
